include/mitsuba/core/distr_1d.h

```diff
@@ -472,7 +502,41 @@ template <typename Value> struct ContinuousDistribution {
     }
 
 private:
-    void compute_cdf(const ScalarFloat *pdf, size_t size) {
+    void compute_cdf() {
+        static_assert(dr::is_jit_v<Float>,
```

Comment:
I think this is technically not legal. You can static_assert on something that is a template parameter of a function. But when it's a template parameter of the class, then that function is in a sense already instantiated when the class is instantiated (since it's not a template function). It's probably just laziness in the compiler that this static assert isn't triggered.